theano/sandbox/gpuarray/type.py

```diff
@@ -751,3 +758,34 @@ def c_code_cache_version(self):
 declaration of an operation.
 """
 gpu_context_type = GpuContextType()
+
+
+# THIS WORKS But CudaNdarray instances don't compare equal to one
```

Comment:
Also, numpy arrays do the same thing.  So I'm pretty sure the cmodule cache has tricks to not save arrrays.